pkg/slacknotifier/slack_oauth.go

```diff
@@ -1,52 +1,98 @@
 package slacknotifier
 
 import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"net/http"
+	"time"
 
 	"github.com/gofiber/fiber/v2"
-	"golang.org/x/oauth2"
 )
 
+const slackAccessUrl = "https://slack.com/api/oauth.v2.access"
+
 var (
 	SlackBotClientID     = ""
 	SlackBotClientSecret = ""
-	oauthConfig          = &oauth2.Config{
-		ClientID:     SlackBotClientID,
-		ClientSecret: SlackBotClientSecret,
-		Endpoint: oauth2.Endpoint{
-			AuthURL:  "https://slack.com/oauth/v2/authorize",
-			TokenURL: "https://slack.com/api/oauth.v2.access",
-		},
-		Scopes: []string{
-			"chat:write",
-			"chat:write.public"},
-	}
 )
 
+type oauthResponse struct {
+	Ok         bool   `json:"ok"`
+	AppID      string `json:"app_id"`
+	AuthedUser struct {
+		ID string `json:"id"`
+	} `json:"authed_user"`
+	Scope       string `json:"scope"`
+	TokenType   string `json:"token_type"`
+	AccessToken string `json:"access_token"`
+	BotUserID   string `json:"bot_user_id"`
+	Team        struct {
+		ID   string `json:"id"`
+		Name string `json:"name"`
+	} `json:"team"`
+	Enterprise          interface{} `json:"enterprise"`
+	IsEnterpriseInstall bool        `json:"is_enterprise_install"`
+}
+
 // OauthHandler creates a handler for slack authentication
 func OauthHandler() fiber.Handler {
 	return func(c *fiber.Ctx) error {
-		ctx := c.Context()
 
-		errStr := c.Params("error")
+		errStr := c.Query("error", "")
 		if errStr != "" {
 			c.Status(http.StatusUnauthorized)
 			_, err := c.WriteString(errStr)
 			return err
 		}
-		code := c.Params("code")
+		code := c.Query("code", "")
 		if code == "" {
 			c.Status(http.StatusBadRequest)
 			_, err := c.WriteString("Code was not provided")
 			return err
 		}
+		//fmt.Printf("\n---->code is |%s|\n", code)
+
+		var slackClient = &http.Client{Timeout: 10 * time.Second}
+
+		req, err := http.NewRequest(http.MethodGet, slackAccessUrl, nil)
+		if err != nil {
+			c.Status(http.StatusInternalServerError)
+			_, err = c.WriteString(fmt.Sprintf("\nFailed to create request: %+v\n", err))
+			return err
```

Comment:
guess, in fiber handler better to use s.Error(c, http.Status, err)

Reply:
@vsukhin this is a separate handler, perhaps we will want to move it to a separate service, is it worth adding it to Server?

Reply:
Yes, I would add it to server. Better to have all hadlers in one place

Reply:
✅